web/assets/js/main_out.js

```diff
@@ -1231,106 +1291,117 @@
+                if (isTyping) {
+                    chatBox.blur();
+                    var chattxt = chatBox.value;
+                    if (chattxt.length > 0) sendChat(chattxt);
+                    chatBox.value = "";
+                } else chatBox.focus();
+                break;
+            case 32: // space
+                if (isTyping || escOverlayShown || pressed.space) break;
+                wsSend(UINT8_CACHE[17]);
+                pressed.space = true;
+                break;
+            case 87: // W
+                if (isTyping || escOverlayShown) break;
+                wsSend(UINT8_CACHE[21]);
+                pressed.w = true;
+                break;
+            case 81: // Q
+                if (isTyping || escOverlayShown || pressed.q) break;
+                wsSend(UINT8_CACHE[18]);
+                pressed.q = true;
+                break;
+            case 69: // E
+                if (isTyping || escOverlayShown || pressed.e) break;
+                wsSend(UINT8_CACHE[22]);
+                pressed.e = true;
+                break;
+            case 82: // R
+                if (isTyping || escOverlayShown || pressed.r) break;
+                wsSend(UINT8_CACHE[23]);
+                pressed.r = true;
+                break;
+            case 84: // T
+                if (isTyping || escOverlayShown || pressed.t) break;
+                wsSend(UINT8_CACHE[24]);
+                pressed.t = true;
+                break;
+            case 80: // P
+                if (isTyping || escOverlayShown || pressed.p) break;
+                wsSend(UINT8_CACHE[25]);
+                pressed.p = true;
+                break;
+            case 27: // esc
+                if (pressed.esc) break;
+                pressed.esc = true;
+                if (escOverlayShown) hideESCOverlay();
+                else showESCOverlay();
+                break;
+        }
+    }
+    function keyup(event) {
+        switch (event.keyCode) {
+            case 32: // space
+                pressed.space = false;
+                break;
+            case 87: // W
+                pressed.w = false;
+                break;
+            case 81: // Q
+                if (pressed.q) wsSend(UINT8_CACHE[19]);
+                pressed.q = false;
+                break;
+            case 69: // E
+                pressed.e = false;
+                break;
+            case 82: // R
+                pressed.r = false;
+                break;
+            case 84: // T
+                pressed.t = false;
+                break;
+            case 80: // P
+                pressed.p = false;
+                break;
+            case 27: // esc
+                pressed.esc = false;
+                break;
+        }
+    }
+    function handleScroll(event) {
+        if (event.target !== mainCanvas) return;
+        mouseZ *= event.deltaY > 0 ? 0.8 : 1.2;
+        mouseZ = Math.max(mouseZ, settings.moreZoom ? 0.08 : 1);
```

Comment:
This is too low! I suggest 0.3....

Reply:
What's the harm in having it low?

Reply:
lag, zoom glitches on trackpads

Reply:
There is no extra lag if you zoom out the same amount. I'm not sure what glitch with trackpads you're talking about, maybe create an issue?